File: pkg/utils/id/idGenerator_test.go
package id

import (
	"github.com/stretchr/testify/assert"
	"testing"
	"time"
)

func TestSnowFlakeIDGenerator(t *testing.T) {
	expect := SnowFlakeID("6733130380397772800")
	time, _ := time.Parse(time.RFC3339, "2020-12-02T15:04:05+09:00")
	generator := Generator(&SnowFlakeIDGenerator{Time: time})
	assert.Equal(t, expect, generator.NewID())
}

File: src/helpers/id_generator.test.ts
import { assert, describe, it } from "vitest";
import { SnowflakeIDGenerator } from "./id_generator.js";

const generator = new SnowflakeIDGenerator(0);

describe("ID生成ができる", () => {
  it("複数回生成してもIDが重複しない", () => {
    let oldID = "";
    for (let i = 0; i < 10000; i++) {
      const newID = generator.generate();
      assert.notEqual(newID, oldID);
      oldID = newID;
    }
  });
});
